internal/syncer/github_pr_commits.go

```diff
@@ -0,0 +1,153 @@
+package syncer
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/mergestat/fuse/internal/db"
+	uuid "github.com/satori/go.uuid"
+)
+
+const selectGitHubPRCommits = `SELECT github_prs.number AS pr_number, github_pr_commits.* FROM github_prs(?), github_pr_commits(?, github_prs.number)`
+
+type githubPRCommit struct {
+	PRNumber       *int       `db:"pr_number"`
+	Hash           *string    `db:"hash"`
+	Message        *string    `db:"message"`
+	AuthorName     *string    `db:"author_name"`
+	AuthorEmail    *string    `db:"author_email"`
+	AuthorWhen     *time.Time `db:"author_when"`
+	CommitterName  *string    `db:"committer_name"`
+	CommitterEmail *string    `db:"committer_email"`
+	CommitterWhen  *time.Time `db:"committer_when"`
+	Additions      *int       `db:"additions"`
+	Deletions      *int       `db:"deletions"`
+	ChangedFiles   *int       `db:"changed_files"`
+	URL            *string    `db:"url"`
+}
+
+// sendBatchGitHubPRCommits uses the pg COPY protocol to send a batch of GitHub pr commits
+func (w *worker) sendBatchGitHubPRCommits(ctx context.Context, tx pgx.Tx, j *db.DequeueSyncJobRow, batch []*githubPRCommit) error {
+	cols := []string{
+		"repo_id",
+		"pr_number",
+		"hash",
+		"message",
+		"author_name",
+		"author_email",
+		"author_when",
+		"committer_name",
+		"committer_email",
+		"committer_when",
+		"additions",
+		"deletions",
+		"changed_files",
+		"url",
+	}
+
+	duplicated := make(map[string]bool)
+
+	inputs := make([][]interface{}, 0, len(batch))
+	for _, r := range batch {
+		var repoID uuid.UUID
+		var err error
+		if repoID, err = uuid.FromString(j.RepoID.String()); err != nil {
+			return fmt.Errorf("uuid: %w", err)
+		}
+
+		key := fmt.Sprintf("%d-%s", *r.PRNumber, *r.Hash)
+		if duplicated[key] {
+			w.logger.Warn().Msgf("duplicated input: %s, %s", repoID.String(), key)
+			continue
+		} else {
+			duplicated[key] = true
+		}
```

Comment:
let's remove this duplicated check for now for consistency with the other handlers. I will make a note, in the future I think we can try a technique like so: https://stackoverflow.com/questions/48019381/how-postgresql-copy-to-stdin-with-csv-do-on-conflic-do-update (copy into a temp table, then insert from there) to ignore any duplicates gracefully. This could be a better way in general to do the batch inserts

Reply:
done